L1Trigger/TrackFindingTracklet/interface/imath.h

```diff
@@ -0,0 +1,1122 @@
+//                   adjust variable shift so the K is as close to Knew as possible (needed for bit length adjustments)
+//                   if do_assert is true, throw an exeption if Knew/Kold is not a power of two
+//                   epsilon is a comparison precision, nbits forces the bit length (possibly discarding MSBs)
+//
+// var_adjustKR (string name, var_base *p1, double Knew, double epsilon = 1e-5, bool do_assert = false, int nbits = -1)
+//                   - same as adjustK(), but with rounding, and therefore latency=1
+//
+// bool calculate(int debug_level) runs through the entire formula tree recalculating both ineteger and floating point values
+//                     returns true if everything is OK, false if obvious problems with the calculation exist, i.e
+//                                  -  integer value does not fit into the alotted number of bins
+//                                  -  integer value is more then 10% or more then 2 away from fval_/K_
+//                     debug_level:  0 - no warnings
+//                                   1 - limited warning
+//                                   2 - as 1, but also include explicit warnings when LUT was used out of its range
+//                                   3 - maximum complaints level
+//
+// var_flag (string name, var_base *cut_var, var_base *...)
+//
+//                    flag to apply cuts defined for any variable. When output as Verilog, the flag
+//                    is true if and only if the following conditions are all true:
+//                       1) the cut defined by each var_cut pointer in the argument list must be passed
+//                       by the associated variable
+//                       2) each var_base pointer in the argument list that is not also a var_cut
+//                       pointer must pass all of its associated cuts
+//                       3) all children of the variables in the argument list must pass all of their
+//                       associated cuts
+//                    The var_flag::passes() method replicates the behavior of the output Verilog,
+//                    returning true if and only if the above conditions are all true. The
+//                    var_base::local_passes() method can be used to query if a given variable passes
+//                    its associated cuts, regardless of whether its children do.
+//
+#ifndef L1Trigger_TrackFindingTracklet_interface_imath_h
+#define L1Trigger_TrackFindingTracklet_interface_imath_h
+
+//use root if uncommented
+#ifndef CMSSW_GIT_HASH
+#define IMATH_ROOT
+#endif
+
+#include <limits>
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <map>
+#include <cmath>
+#include <sstream>
+#include <string>
+#include <cassert>
+#include <set>
+
+#include "L1Trigger/TrackFindingTracklet/interface/Util.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#ifdef IMATH_ROOT
+#include "TH2F.h"
```

Comment:
can you use
```suggestion
#include <TH2F.h>
```
*etc*. ?

Reply:
fixed as suggested